### DISCUSSION TITLE
How to get EIP1193 Provider from ` privateKeyToAccount`

### DISCUSSION THREAD

**Discussion by technophile-04** [wevm/viem#1988] [Question]
We are trying to implement Connector for wagmi v2, where we need to create `getProvider`. We wanted the Connector to be linked with account created from ` privateKeyToAccount`. 

Here is the naive example : 

```ts
export const createBurnerConnector = () => {
  //   ...
  return createConnector<Provider>(config => ({
  // ...
      async getProvider({ chainId } = {}) {
      const chain = config.chains.find(x => x.id === chainId) ?? config.chains[0];
      const burnerAccount = privateKeyToAccount(loadBurnerSK());
      const client = createWalletClient({
        chain: chain,
        account: burnerAccount,
        transport: http(),
      });

      const request: EIP1193RequestFn = async (...args) => {
        // @ts-expect-error
        return await client.request(...args);
      };

      return custom({ request })({ retryCount: 0 });
    }, 
 
  // ...
  }));
};
```

But the problem with above approach is since we are using `http` transport its always getting connected to Hardhat's 0th account because on local network I think `eth_account` to http://127.0.0.1:8545/ gives hardhat's 0th account. 

how do I tell `createWalletClient` to use passed in account for all the methods ?

**Comment by tmm** [comment]
The `mock` connector in Wagmi might be helpful to take a look at

https://github.com/wevm/wagmi/blob/main/packages/core/src/connectors/mock.ts

**Reply by technophile-04** [reply]
Yup thanks to `mock` connector example we are doing something similar already : 

https://github.com/scaffold-eth/scaffold-eth-2/blob/6c5dbed60887b943f5810cc9553bcdf9220cbecc/packages/nextjs/services/web3/wagmi-burner/createBurnerConnector.ts#L83

Was just wondering if there was more easy / robust way to do it instead of kind of creating our own `EIP1193RequestFn`. 

But tysm 🙌
